src/foremast/pipeline/construct_pipeline_block.py

```diff
@@ -179,14 +179,19 @@ def ec2_pipeline_setup(appname='', project='', settings=None, env='', region='',
 
     Returns:
         dict: Updated settings to pass to templates for EC2 info
-    """
 
+    """
     data = copy.deepcopy(settings)
-    user_data = generate_encoded_user_data(env=env, region=region, app_name=appname, group_name=project)
+    user_data = generate_encoded_user_data(
+        env=env,
+        region=region,
+        generated=generated,
+        group_name=project,
+    )
 
     # Use different variable to keep template simple
     instance_security_groups = sorted(DEFAULT_EC2_SECURITYGROUPS[env])
-    instance_security_groups.append(appname)
+    instance_security_groups.append(generated.app_name())
```

Comment:
Do you think there is a way to change this to another `Generator()` object? This is for cases where the default group name doesn't match the appname.

Reply:
Where would this different Group name come from?

Reply:
It would come from `gogoutilities.Generator()`. We could add another facility for `elb` and default it to the format used for `appname`. Just a thought.

Reply:
It would come from `gogoutilities.Generator()`. We could add another facility for `elb` and default it to the format used for `appname`. Just a thought.

Reply:
It sounds like you want `gogoutils.Generator.security_group` and `gogoutils.Generator.elb`. This could be achieved with the following and avoid another Instance of `Generator`.
```python
class Generator:
    def __init__(self):
        self.data = {
            'elb': '{app}',
            'security_group': '{app}',
        }


Generator(formats={
    'elb': 'custom_elb_name',
    'security_group': 'custom_name',
})
```

Reply:
The changes that i was thinking were something like this: https://github.com/foremast/gogo-utils/pull/17

Reply:
Updated from foremast/gogo-utils#17.